client/data/geo/test/use-geolocation-query.test.tsx

```diff
@@ -0,0 +1,58 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { renderHook, waitFor } from '@testing-library/react';
+import { when } from 'jest-when';
+import { act, ReactNode } from 'react';
+import { useGeoLocationQuery, GeoLocationData } from '../use-geolocation-query';
+
+jest.useFakeTimers();
+
+describe( 'useGeoLocationQuery', () => {
+	const wrapper = ( { children }: { children: ReactNode } ) => (
+		<QueryClientProvider client={ new QueryClient() }>{ children }</QueryClientProvider>
+	);
+
+	const mockGeoData: GeoLocationData = {
+		city: 'San Francisco',
+		country_long: 'United States',
+		country_short: 'US',
+		latitude: '37.7749',
+		longitude: '-122.4194',
+		region: 'California',
+	};
+
+	beforeAll( () => {
+		when( jest.spyOn( globalThis, 'fetch' ) )
```

Comment:
A bummer to introduce yet another way to mock requests.

However, I'm not opposed, since I find this syntax readable and simpler than `nock`. 

Could be an opportunity for refactoring at some point.

Reply:
> A bummer to introduce yet another way to mock requests.
> 
> However, I'm not opposed, since I find this syntax readable and simpler than `nock`.
> 
> Could be an opportunity for refactoring at some point.

Yeah, I'd much rather use `nock` than stubbing `fetch` like this, though I think `jest-when` makes it at least somewhat readable to mock responses to specific arguments. I'm not sure if / when the JSDOM Jest environment will support native fetch properly, but elsewhere (e.g. #103054), I'd continue to prefer Nock where it's available (the default, i.e. non-JSDOM Jest environment supports it).

Reply:
> Yeah, I'd much rather use `nock` than stubbing fetch like this

For example, this mock would fail if someone called fetch like `fetch( 'https://public-api.wordpress.com/geo/', { method: 'get' } );` even though it's functionally equivalent.